### PR TITLE
Reword and extend list of supported SQL statements

### PR DESCRIPTION
**Pull request by jhlodin**
Reword the supported SQL statements to be more explicit about which are supported. Extend list out with more statements, with intent to pare down and explain as needed.

See https://github.com/trinodb/trino/pull/8419 for similar work on the PostgreSQL connector.

### CODE REVIEW

File: docs/src/main/sphinx/connector/sqlserver.rst [hunk 1]
```diff
@@ -114,14 +114,66 @@ Complete list of `SQL Server data types
...
+* :doc:`/sql/create-table`
+* :doc:`/sql/create-table-as`
+* :doc:`/sql/drop-schema`
+* :doc:`/sql/drop-table`
+* :doc:`/sql/insert`
+* :doc:`/sql/show-create-schema`
+* :doc:`/sql/show-create-table`
+* :doc:`/sql/update`
+
+Security:
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/sqlserver.rst:154]: I dont think the connector supports any security stuff

**Review by m57lyra — APPROVED** [review]
LGTM

File: docs/src/main/sphinx/connector/sqlserver.rst [hunk 2]
```diff
@@ -114,14 +114,66 @@ Complete list of `SQL Server data types
...
+
+Write access:
+
+* :doc:`/sql/create-schema`
+* :doc:`/sql/create-table`
+* :doc:`/sql/create-table-as`
+* :doc:`/sql/drop-schema`
+* :doc:`/sql/drop-table`
+* :doc:`/sql/insert`
+* :doc:`/sql/show-create-schema`
```
Comment (hashhar) [inline docs/src/main/sphinx/connector/sqlserver.rst:150]: Not supported.

File: docs/src/main/sphinx/connector/sqlserver.rst [hunk 3]
```diff
@@ -114,14 +114,66 @@ Complete list of `SQL Server data types
...
+* :doc:`/sql/show-create-table`
+* :doc:`/sql/update`
```
Comment (hashhar) [inline docs/src/main/sphinx/connector/sqlserver.rst:152]: Not supported except limited circumstances? cc: @kokosing 

```
trino:guest> update nation set name = 'ashhar';
Query 20210708_053555_00031_733ea failed: This connector does not support updates
```

File: docs/src/main/sphinx/connector/sqlserver.rst (hunk 1 above)
Comment (hashhar) [reply docs/src/main/sphinx/connector/sqlserver.rst:154]: Correct. Roles are only fully supported in the Hive connector AFAIK. But please check with Dain maybe.

**Review by hashhar — COMMENTED** [review]
Some comments.

File: docs/src/main/sphinx/connector/sqlserver.rst [hunk 4]
```diff
@@ -114,14 +114,66 @@ Complete list of `SQL Server data types
...
+* :doc:`/sql/show-tables`
+* :doc:`/sql/describe`
+
+Write access:
+
+* :doc:`/sql/create-schema`
+* :doc:`/sql/create-table`
+* :doc:`/sql/create-table-as`
+* :doc:`/sql/drop-schema`
+* :doc:`/sql/drop-table`
```
Comment (hashhar) [inline docs/src/main/sphinx/connector/sqlserver.rst:148]: Requires setting `allow-drop-table` in the JDBC connector config.

**Review by rosewms — APPROVED** [review]

File: docs/src/main/sphinx/connector/sqlserver.rst (hunk 4 above)
Comment (jhlodin) [reply docs/src/main/sphinx/connector/sqlserver.rst:148]: I believe Jessie212 is planning to add that info

Comment (Ordinant) [reply docs/src/main/sphinx/connector/sqlserver.rst:149]: I'm confused by this resolution. @Jessie212 will add the fact that the drop-table command requires the allow-drop-table=true config setting ... where? In this file? When? I hope not in another file. This appears to be the right place to say that this command is supported but requires a setting to activate it.

**Review by Ordinant — APPROVED** [review]
Approved overall, but a comment left about something confusing. I am easily confused ...

File: docs/src/main/sphinx/connector/sqlserver.rst (hunk 4 above)
Comment (jhlodin) [reply docs/src/main/sphinx/connector/sqlserver.rst:149]: My 2 cents, what I've seen in other supported SQL statements lists is a ref link here to a section down below in the same page that explains in more detail. So I'd expect it to be the same file, but maybe a standardized blurb between each JDBC connector for consistency.

Comment (Jessie212) [reply docs/src/main/sphinx/connector/sqlserver.rst:149]: @Ordinant, @jhlodin is correct. I'm going to include it as a fragment.

**Review by hashhar — APPROVED** [review]

**Pull request merged**